src/hotspot/share/runtime/threads.cpp

```diff
@@ -1331,9 +1331,11 @@ void Threads::print_on(outputStream* st, bool print_stacks,
         if (thread_oop != nullptr) {
           if (p->is_vthread_mounted()) {
             const oop vt = p->vthread();
-            assert(vt != nullptr, "vthread should not be null when vthread is mounted");
-            st->print_cr("   Mounted virtual thread \"%s\" #" INT64_FORMAT, JavaThread::name_for(vt), (int64_t)java_lang_Thread::thread_id(vt));
-            p->print_vthread_stack_on(st);
+            if (vt != thread_oop) {
```

Comment:
We need a comment explaining why we have to check for this as without knowing the implementation quirks it seems non-sensical for a thread to have mounted itself!

Reply:
> We need a comment explaining why we have to check for this as without knowing the implementation quirks it seems non-sensical for a thread to have mounted itself!

Just to note that JavaThread._vthread is always the "current thread". It's the virtual thread when mounted, the carrier (as in itself) when there is no virtual thread mounted. It's one this way so that the C2 instrinic for currentThread doesn't need a branch/test.

Reply:
I wasn't aware of that, and I was going to suggest to add that information as documentation, but it seems to be already documented here: https://github.com/openjdk/jdk/blob/ad0b54d429fdbd806c09aa06bb42f1ed4a0297e8/src/hotspot/share/runtime/javaThread.hpp#L95

Reply:
I've still clarified why the if is required in 9a954efe164d9caee31494773c0eecc8ff69223f